### PR TITLE
Report mismatched translation blob in Features

### LINKED ISSUE #3543: Report "outdated translation blob" to host
this should probably be a new `Features` field? so that Suite can detect this and silently update to matching version.

note that with current code, it doesn't matter which direction the version goes, silent upgrade as well as downgrade is allowed. This seems right in that we ideally want a translation blob matching the firmware version. (e.g., in newer blobs, some older strings may be omitted)

### PR DESCRIPTION
**Pull request by mmilata**
Also returns False when blob is newer than firmware. Comparison includes build version (as in major.minor.patch.build) which we don't report in Features, not completely sure if that's right.

Fixes #3543.

### CODE REVIEW

**Comment by matejcik** [comment]
there's a bug, tests are broken.
(i don't see the bug from the code though)

File: core/src/apps/base.py [hunk 1]
```diff
@@ -43,6 +43,20 @@ def busy_expiry_ms() -> int:
...
+def _language_version_matches() -> bool | None:
+    """
+    Whether translation blob version matches firmware version.
+    Returns None if there is no blob.
+    """
+    from trezor import translations
+
+    header = translations.TranslationsHeader.load_from_flash()
+    if header is None:
+        return None
```
Comment (matejcik) [inline core/src/apps/base.py:55]: I would lean towards returning `True` here
semantically, "language is en-US and it does not need update"

File: common/protob/messages-management.proto [hunk 2]
```diff
@@ -131,6 +131,7 @@ message Features {
     optional uint32 homescreen_width = 47;         // homescreen width in pixels
     optional uint32 homescreen_height = 48;         // homescreen height in pixels
     optional bool bootloader_locked = 49;       // bootloader is locked
+    optional bool language_version_matches = 50;   // translation blob version matches firmware version
```
Comment (matejcik) [inline common/protob/messages-management.proto:134]: perhaps even `default=true` to resolve old models that don't send this field?

that way Suite can just `if (!language_version_matches) { sendLanguageBlob(...) }` in exactly the situations when required?
cc @mroz22 what do you think?

File: core/src/apps/base.py (hunk 1 above)
Comment (mmilata) [reply core/src/apps/base.py]: Fixed in [a6dfe95](https://github.com/trezor/trezor-firmware/pull/3544/commits/a6dfe9558cdcdba2723d39a7b59b90fe1674e7b5), this was actually the reason for the failing test.

Comment (matejcik) [reply core/src/apps/base.py]: but now you also need to update the test :)

Comment (mmilata) [reply core/src/apps/base.py]: uhhhh yeah 3215107b2da53aba9207586677f55404f3443c19

File: common/protob/messages-management.proto (hunk 2 above)
Comment (mmilata) [reply common/protob/messages-management.proto]: added in 7abca4e5ec0e4f4072d0d6227cd94b2d6ec90fa4

**Review by matejcik — APPROVED** [review]

**Pull request merged**